src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/DataSourceParser.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient.SNI;
+using Microsoft.Data.SqlClientX.Handlers;
+
+namespace Microsoft.Data.SqlClient.Microsoft.Data.SqlClientX.Handlers
+{
+    /// <summary>
+    /// Handler to populate data source information required for transport handler
+    /// </summary>
+    internal class DataSourceParser : IHandler<ConnectionHandlerContext>
```

Comment:
Can the class be called `DataSourceParsingHandler` ? 
I am hoping to look at the class name and know that it was a handler.

Reply:
Done